File: src/springboot-mongodb/src/main/java/org/jrue/springbootmongodb/controller/UserController.java
package org.jrue.springbootmongodb.controller;

import org.jrue.springbootmongodb.model.User;
import org.jrue.springbootmongodb.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/users")
public class UserController {

    @Autowired
    private UserRepository userRepository;

    @GetMapping
    public List<User> getAll() {
        return userRepository.findAll();
    }

    @PostMapping
    public String addDocument(@RequestBody User user) {
        userRepository.save(user);
        return "User " + user.getFirstName() + " has been saved";
    }
}

File: src/springboot-eureka-client2/src/main/kotlin/org/jrue/springbooteurekaclient2/UserController.kt
package org.jrue.springbooteurekaclient2

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/users")
class UserController {


    data class Person(val userId: Int, val name: String);

    @GetMapping("/{userId}")
    fun getUser(@PathVariable userId: String) : Person {

       val users = mapOf(123 to Person(123,"John"), 124 to Person(124,"Doe"),
               125 to Person(125,"Michael"), 126 to Person(126,"Joel"));
        println("UserId ${userId} users: ${users} ${users.getOrDefault(userId.toInt(), Person(0,"Name Not found"))}")
       return users.getOrDefault(userId.toInt(), Person(0,"Name Not found"));
    }

}